src/Rules/DSL/ThatParser.php

```diff
@@ -7,5 +7,7 @@
 
 interface ThatParser
 {
+    public function exclude(string ...$classesToBeExcluded): self;
```

Comment:
Should we define a new interface for the DSL? Placing that method here allow also this call chain

```
allClasses()
->exclude('Banana')
->that(ResideInaNamespace('Fruit')
->exclude('Mango')
->that(ImplementInterface('HasVitaminC'))
...
```

To me such syntax is misleading, since the call to `->exclude('Mango')` is ambiguous: does it exclude classes from the whole class set or only from the first `->that(ResideInaNamespace('Fruit')` invocation?

For this reason I think ->exclude should be callable right after allClasses() and in no other places

Reply:
You can't use this syntax because `that` returns `AndThatShouldParser` interface.
So you can't use `except` after `that`